x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/Neg.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic;
+
+import org.elasticsearch.compute.ann.Evaluator;
+
+public class Neg {
+    @Evaluator(extraName = "Ints", warnExceptions = { ArithmeticException.class })
+    static int processInts(int v) {
+        return Math.negateExact(v);
+    }
+
+    @Evaluator(extraName = "Longs", warnExceptions = { ArithmeticException.class })
+    static long processLongs(long v) {
+        return Math.negateExact(v);
+    }
+
+    @Evaluator(extraName = "Doubles")
+    static double processDoubles(double v) {
+        return -v;
```

Comment:
Need to check for overflow negation here as well:
```
if (v == Double.MAX_VALUE || v == Double.MIN_VALUE) {
     throw new ArithmeticOverflow("double overflow");
}
```

Reply:
I guess this should be fine, no? It's just a bit flip.

Reply:
That was my thinking as well. Anyway, I'll double check that and at least add a comment explaining why this is fine, if correct.

Reply:
Just manually tested negating `Double.MAX_VALUE` (and its negative), which is fine.

I don't see a good way to add a test case for this:
- wouldn't trust a csv test to correctly construct `Double.MAX_VALUE` since there may be rounding errors when parsing the literal)
- there's no existing test class where this would fit in, and it feels odd to add a new test class just for this.

So, if we're happy with that, I'd like to simply add a comment here that explains that negating a double always fine.

Reply:
Added unit tests for the Neg evaluators, also covering all edge cases.